api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2745,12 +2809,13 @@ public void composeObject(
         }
       }
 
-      completeMultipartUpload(bucketName, objectName, uploadId, totalParts);
+      return completeMultipartUpload(
+          args.bucket(), getRegion(args.bucket()), args.object(), uploadId, totalParts, null, null);
     } catch (RuntimeException e) {
-      abortMultipartUpload(bucketName, objectName, uploadId);
+      abortMultipartUpload(args.bucket(), args.object(), uploadId);
       throw e;
     } catch (Exception e) {
-      abortMultipartUpload(bucketName, objectName, uploadId);
+      abortMultipartUpload(args.bucket(), args.object(), uploadId);
```

Comment:
`abortMultipartUpload()` should be called  not `if (args.sources.size() == 1 && src.offset() == null && src.length() == null) {`